src/control/lib/spdk/src/nvme_control.c

```diff
@@ -70,6 +70,67 @@ nvme_discover(void)
 	return _discover(&spdk_nvme_probe, true, &get_dev_health_logs);
 }
 
+struct ret_t *
+nvme_wipe_first_ns(char *ctrlr_pci_addr)
+{
+	int			 ns_id, rc;
+	struct spdk_nvme_ns	*ns;
+	struct spdk_nvme_qpair	*qpair;
+	struct ctrlr_entry	*ctrlr_entry;
+	struct spdk_nvme_ctrlr	*ctrlr;
+	spdk_nvme_cmd_cb	 cb_fn = NULL;
+	void			*cb_arg = NULL;
+	struct ret_t		*ret;
+
+	ret = init_ret(0);
+
+	ret->rc = get_controller(&ctrlr_entry, ctrlr_pci_addr);
+	if (ret->rc != 0)
+		return ret;
+
+	ctrlr = ctrlr_entry->ctrlr;
+
+	for (ns_id = spdk_nvme_ctrlr_get_first_active_ns(ctrlr);
+	     ns_id != 0;
+	     ns_id = spdk_nvme_ctrlr_get_next_active_ns(ctrlr, ns_id)) {
+		ns = spdk_nvme_ctrlr_get_ns(ctrlr, ns_id);
+		assert(ns != NULL);
+	}
+
+	if (ns_id != 1) {
+		/* unexpected, don't wipe */
+		snprintf(ret->err, sizeof(ret->err),
+			"first active namespace has ID %d, expected 1\n",
+			ns_id);
+		ret->rc = -NVMEC_ERR_NS_ID_UNEXPECTED;
+		return ret;
+	}
+
+	qpair = spdk_nvme_ctrlr_alloc_io_qpair(ctrlr, NULL, 0);
+	if (qpair == NULL) {
+		snprintf(ret->err, sizeof(ret->err),
+			"spdk_nvme_ctrlr_alloc_io_qpair() failed\n");
+		ret->rc = -NVMEC_ERR_ALLOC_IO_QPAIR;
+		return ret;
+	}
+
+	rc = spdk_nvme_ns_cmd_write_zeroes(ns, qpair,
+		0, /* LBA start */
+		1, /* number of LBAs */
+		cb_fn, cb_arg, 0);
+	if (rc != 0) {
+		snprintf(ret->err, sizeof(ret->err),
+			"starting write I/O failed (rc: %d)", rc);
+		ret->rc = -NVMEC_ERR_NS_WRITE_FAIL;
+		return ret;
+	}
+
+	spdk_nvme_qpair_process_completions(qpair, 0);
+	spdk_nvme_ctrlr_free_io_qpair(qpair);
```

Comment:
Does it somehow still do a trim of the overall device?

Reply:
this wipes the structure signatures from LBA-0 by writing zeros so , yes. no blobs are discovered after